File: codeDaily/CS97SI/HardProblem/HyperKnight/solve/cpp/HyperKnightRunner.cpp
#include <topcoder>
using namespace TopCoder;
#include "HyperKnight.cpp"
#include <fstream>
using namespace std;

int main(int argc, char const *argv[]) {
    try {
        ifstream ifs(argv[1]);
        int a; read(ifs, a); next(ifs);
        int b; read(ifs, b); next(ifs);
        int numRows; read(ifs, numRows); next(ifs);
        int numColumns; read(ifs, numColumns); next(ifs);
        int k; read(ifs, k);
        ifs.close();

        ofstream ofs(argv[2]);
        show(ofs, countCells(a, b, numRows, numColumns, k));
        ofs.close();
    } catch (exception &e) {
        cerr << e.what() << endl;
    }
    return 0;
}

File: codeDaily/CS97SI/HardProblem/HyperKnight/solve/haskell/HyperKnightRunner.hs
import TopCoder
import System.Environment (getArgs)
import System.IO
import HyperKnight

getVars :: Parser (Int, Int, Int, Int, Int)
getVars = do a <- spaces >> parseInt ; next
             b <- spaces >> parseInt ; next
             numRows <- spaces >> parseInt ; next
             numColumns <- spaces >> parseInt ; next
             k <- spaces >> parseInt
             return (a, b, numRows, numColumns, k) 

main = do 
    args <- getArgs
    hIn <- openFile (head args) ReadMode
    contents <- hGetContents hIn
    case (parse getVars "variables" contents) of
        Left err -> print err
        Right (a, b, numRows, numColumns, k) -> do
            hOut <- openFile (head (tail args)) WriteMode
            hPutStr hOut $ show $ countCells a b numRows numColumns k
            hClose hOut
    hClose hIn
